editor/components/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -71,10 +71,29 @@ class HierarchicalTermSelector extends Component {
 
 	onAddTerm( event ) {
 		event.preventDefault();
-		const { formName, formParent, adding } = this.state;
+		const { onUpdateTerms, restBase, terms, slug } = this.props;
+		const { formName, formParent, adding, availableTerms } = this.state;
 		if ( formName === '' || adding ) {
 			return;
 		}
+
+		// check if the term we are adding already exists
+		const existingTerm = find( availableTerms, term => {
```

Comment:
Noting that this doesn't fix the issue really because this will only search in the availableTerms locally which doesn't correspond to the whole list of terms.

I personally think we can't reliably fix this issue right now because there's no API to fetch the terms by "name" and "parent" (search is not perfect and could return results even if it's not a duplicate).

Reply:
This avoids making the request, if this fails and the category is not on the list, then we try to add if the server returns a 409 we are totally certain that a duplicate exists than we search for it, and we use the return of the search. The only case where we may fail is the search returns more than one term besides the duplicate, I will apply a client-side filter by name and parent on the search result this way we are certain we handle all cases.